### PR TITLE
Update test-infra to use new Interface instead of CommonEnvironment

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Fix all the tests to match the refactor done in https://github.com/DataDog/test-infra-definitions/pull/688

We no longer need to explicitly pass the `CommonEnvironment` for component that are cloud agnostic. It now uses an `Env` interface that must be implement by any Environment (`AWS`, `Azure`, ...)

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by lrbhatia — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]
LGTM for ASC owned files

**Review by fisherevans — APPROVED** [review]

**Review by jmw51798 — APPROVED** [review]

**Review by derekwbrown — APPROVED** [review]
Approved for wkit on the one file listed that we own.

**Review by FlorentClarret — APPROVED** [review]

File: test/new-e2e/go.mod [hunk 1]
```diff
@@ -29,7 +29,7 @@ require (
 	// `TEST_INFRA_DEFINITIONS_BUILDIMAGES` matches the commit sha in the module version
 	// Example: 	github.com/DataDog/test-infra-definitions v0.0.0-YYYYMMDDHHmmSS-0123456789AB
 	// => TEST_INFRA_DEFINITIONS_BUILDIMAGES: 0123456789AB
-	github.com/DataDog/test-infra-definitions v0.0.0-20240502112654-5942f13180e5
+	github.com/DataDog/test-infra-definitions v0.0.0-20240506124150-49e7ac0452ca
```
Comment (pducolin) [inline test/new-e2e/go.mod:32]: **🥜 nitpick**
Missed the update to `TEST_INFRA_DEFINITIONS_BUILDIMAGES`

**Review by carlosroman — APPROVED** [review]

**Pull request merged**